library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/ViewUtils.kt

```diff
@@ -43,4 +43,16 @@ internal class ViewUtils {
                         .isAssignableFrom(view::class.java)
                 )
     }
+
```

Comment:
@reviewers : The only reason I can't make this class a Singleton is that it uses a state based on the `Build.Version` value when generating the lazy sets for `isSystemView` ids. This breaks my unit tests as once this class is instantiated those `sets` are generated only for the current state of the `Build.Version` so I cannot re - initialize it. I could move those sets inside the `ViewUtils#checkIfSystemNoise` methods but this is called for each view in the tree every time we take a snapshot so there will be too many sets generated. If you have a better idea let me know, if not I don't mind keeping it like that for now.